### PR TITLE
Link Generator and ASPNETCore 2.2

### PR DESCRIPTION
**Pull request by ctolkien**

### CODE REVIEW

**Comment by ctolkien** [comment]
One of the benefits of swapping to this, is that it now follows options configured for `routing`, e.g if you have configured routing to generate lowercase routes, then the sitemap will now follow.

**Comment by PureKrome** [comment]
LGTM. (though I'm not skilled up with Razor stuff)

NitPick: missing null checks in various methods.
NitPick: could also make `Main` an async method.

File: SodaPop.RazorPagesSitemap.Tests/SodaPop.RazorPagesSitemap.Tests.csproj [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+    <TargetFramework>netcoreapp2.2</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.0.1" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0" />
-    <PackageReference Include="xunit" Version="2.3.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1" />
-    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" />
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.2.0" />
```
Comment (poke) [inline SodaPop.RazorPagesSitemap.Tests/SodaPop.RazorPagesSitemap.Tests.csproj]: Also add `<PackageReference Include="Microsoft.AspNetCore" />`

**Comment by poke** [comment]
@PureKrome 
> NitPick: missing null checks in various methods.

Yeah, the library needs some general polishing in some places.

> NitPick: could also make `Main` an async method.

I don’t mind the non-async, considering that’s still what the default template generates.

**Comment by ctolkien** [comment]
>This thing really needs some actual tests btw…
>Yeah, the library needs some general polishing in some places.

PR's very welcome :D

**Pull request merged**